publish.sh

```diff
@@ -0,0 +1,68 @@
+
+# Prerequisites
+if [ "$#" -ne 2 ]; then
+    echo "Illegal number of parameters. Plugin id and version should be passed."
+    exit 1
+fi
+
+PLUGIN_ID=$1
+PLUGIN_VERSION=$2
+
+
+# Login
+oc login ${CHE_OSO_CLUSTER} --insecure-skip-tls-verify=${CHE_OSO_TRUST_CERTS} --token=${CHE_OSO_USER_TOKEN} && oc project ${CHE_OSO_PROJECT}
+
+
+# Start registry if not exist
+REGISTRY=$(oc get svc --field-selector='metadata.name=che-plugin-registry' 2>&1)
+if [[ "$REGISTRY" == "No resources found." ]]; then
+  echo "Creating registry..."
+  oc new-app eclipse/che-plugin-registry > /dev/null
+fi
+
+# Create route if not exist
+ROUTE=$(oc get routes --field-selector='metadata.name=che-plugin-registry' 2>&1)
+if [[ "$ROUTE" == "No resources found." ]]; then
+  echo "Creating route..."
+  oc create route edge --service=che-plugin-registry > /dev/null
+fi
+HOST=$(oc get routes --field-selector='metadata.name=che-plugin-registry'  -o=custom-columns=":.spec.host" | xargs)
+
+# Allow deploy to finish
+echo "Waiting for deploy finish.."
+sleep 10
```

Comment:
I would suggest to:
- always create the route when you create the service
- wait 10s only if the service didn't exist

```bash
if [[ "$REGISTRY" == "No resources found." ]]; then
  echo "Creating registry..."
  oc new-app eclipse/che-plugin-registry > /dev/null
  echo "Creating route..."
  oc create route edge --service=che-plugin-registry > /dev/null
  echo "Waiting for deploy finish.."
  sleep 10
fi
```